client/my-sites/pages/page/index.js

```diff
@@ -203,6 +204,33 @@ class Page extends Component {
 		);
 	}
 
+	setFrontPage() {
+		{
+			alert( 'This feature is still being developed!' );
+		}
+	}
+
+	getFrontPageItem() {
+		if ( this.props.hasStaticFrontPage && this.props.isPostsPage ) {
+			return null;
+		}
+
+		if ( ! utils.userCan( 'edit_post', this.props.page ) ) {
+			return null;
+		}
+
+		if ( ! isEnabled( 'manage/pages/set-front-page' ) ) {
```

Comment:
We could move this as the first check, in order to exit immediately if the feature flag is disabled, skipping the other checks.

Reply:
I agree that it would be nice to move this and make it the first check of `getFrontPageItem` function.